src/main/java/com/robin/hoteless/application/BookingTool.java

```diff
@@ -117,7 +115,16 @@ public static void bookRoom() {
             System.out.println("Your reservation has been confirmed!");
 
             // print reservation confirmation including reservation id
-            newReservation.show();
+            System.out.printf("Your room has been booked under reservation #%s!\n" +
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.